Add post on 2024 developer summit

stefanv opened the pull request:
Hi all,

I've been meaning to get news of the summit out, but then summer happened. Here is a first draft, and I'd be grateful if you can comment on / correct / flesh out the pieces that you were involved with. Don't be shy to push to my branch, I'll do a last round of language editing before posting. 

I just worked through the worklog, so it is quite possible that I missed some work items. Feel free to add them in!

/cc @steppi @arokem @bsipocz @perimosocordiae @drammock @dschult @QuLogic
@larsoner @eriknw @fperez @guenp @InessaPawson @j-bowhay
@jarrodmillman @jakirkham @juanis2112 @ksunden @lagru @munkm @mdhaber
@matthewfeickert @Carreau @melissawm @MridulS @nabobalis @nsmith-
@tupui @ivanov @rossbar @MSanKeys963 @crazy4pi314 @scottyhq @seberg @thomasjpfan @betatim


- [X] The main subject relates to at least one project affiliated to the Scientific Python Ecosystem.
- [X] I have the right to publish the content under BSD 3-Clause License for the code and Creative Common CC-BY-4.0 License for the text.
- [ ] Images have been compressed using a tool like `pngquant`.

(Does JPEG need any further compression?)

tupui reviewed (approved):
This is amazing, thanks for the write up ✨ Concise, punchy, still lots of info 👍 Can't wait to do more of these with you all!!

> JPG
239 KB is ok I would say 👍

mdhaber commented on content/posts/scientific-python/dev-summit-2/index.md:
```diff
@@ -0,0 +1,85 @@
+
+### SciPy
+
+Several of the SciPy developers were present, and we used to opportunity to celebrate Dan Schult joining as core developer. Matt and Pamphile did some work on the new distribution infrastructure, Dan worked on sparse (remotely with CJ), and newly-supported [`const` statements were added to Cython code](https://github.com/scipy/scipy/pull/20891) got reviewed and merged.
```
```suggestion
Several of the SciPy developers were present, and we used the opportunity to celebrate Dan Schult joining as a core developer. Matt and Pamphile did some work on the new distribution infrastructure, Dan worked on sparse (remotely with CJ), and newly-supported [`const` statements were added to Cython code](https://github.com/scipy/scipy/pull/20891) got reviewed and merged.
```

mdhaber commented on content/posts/scientific-python/dev-summit-2/index.md:
```diff
@@ -0,0 +1,85 @@
+
+### Conclusion
+
+Numerous other PRs were made, of which a number were probably not even captured in the [worklog](https://hackmd.io/wsJVTMYdQGG_Zgz7rgxSzw). But, besides the inherent satisfaction of working together with this great group, the best features of the summit was that we were able to hang out, bonding over our communal joys and struggles—both technical and personal.
```
```suggestion
Numerous other PRs were made, of which a number were probably not even captured in the [worklog](https://hackmd.io/wsJVTMYdQGG_Zgz7rgxSzw). But, besides the inherent satisfaction of working together with this great group, the best feature of the summit was that we were able to hang out, bonding over our communal joys and struggles—both technical and personal.
```

mdhaber commented on content/posts/scientific-python/dev-summit-2/index.md:
```diff
@@ -0,0 +1,85 @@
+
+Numerous other PRs were made, of which a number were probably not even captured in the [worklog](https://hackmd.io/wsJVTMYdQGG_Zgz7rgxSzw). But, besides the inherent satisfaction of working together with this great group, the best features of the summit was that we were able to hang out, bonding over our communal joys and struggles—both technical and personal.
+
+We are grateful to the ecosystem developer who gave up their time to attend the summit (many had to put in leave _just to do more work_!). The summits are valuable, and translate to a lot of work work getting done and decisions being made. We hope that there will be more on the horizon!
```
```suggestion
We are grateful to the ecosystem developers who gave up their time to attend the summit (many had to put in leave _just to do more work_!). The summits are valuable, and translate to a lot of work work getting done and decisions being made. We hope that there will be more on the horizon!
```

mdhaber commented on content/posts/scientific-python/dev-summit-2/index.md:
```diff
@@ -0,0 +1,85 @@
+
+This post is a while overdue, but it's been a busy summer for everyone!
+
+In June, several of us got together for the "annual" (well, we've had it twice now) Scientific Python developer summit in Seattle. Our friends at the eScience Institute were again kind enough to host us. This time around, we made the event a bit shorter: a week is a long time to take off, and it also gets a bit overstimulating/exhausting, given the amount of conversations and work that take place ...
```
```suggestion
In June, several of us got together for the "annual" (well, we've had it twice now) Scientific Python developer summit in Seattle. Our friends at the eScience Institute were again kind enough to host us. This time around, we made the event a bit shorter: a week is a long time to take off, and it also gets a bit overstimulating/exhausting, given the amount of work that takes place.
```

stefanv replied:
How about: "given the amount of conversation and work that takes place.".  It's a bit of a cheat, but I think we can get away with it.

mdhaber replied:
Yeah.
```suggestion
In June, several of us got together for the "annual" (well, we've had it twice now) Scientific Python developer summit in Seattle. Our friends at the eScience Institute were again kind enough to host us. This time around, we made the event a bit shorter: a week is a long time to take off, and it also gets a bit overstimulating/exhausting, given the amount of conversation and work that takes place.
```

mdhaber commented on content/posts/scientific-python/dev-summit-2/index.md:
```diff
@@ -0,0 +1,85 @@
+Documentation was a much more popular topic than anticipated!
+
+- The new [mystmd](https://mystmd.org/guide) tooling generated some excitement, and an [experimental port of the NumPy tutorials](https://github.com/numpy/numpy-tutorials/tree/mystjs) was made.
+- A recommendation on [consistent use of backticks was submitted to numpydoc](https://github.com/numpy/numpydoc/pull/525).
```
```suggestion
- Recommendations on consistent use of [backticks](https://github.com/numpy/numpydoc/pull/525) and [monospaced font](https://github.com/pydata/pydata-sphinx-theme/issues/1852) were submitted to numpydoc and pydata-sphinx-theme, respectively.
```

stefanv commented:
If you've left comments, I'll add you as a co-author, unless you object!

nabobalis commented on content/posts/scientific-python/dev-summit-2/index.md:
Just a really and I mean really nitpicky comment. I typically do one sentence per line for text so its easier to diff.

tupui replied:
That's a linter thing 😉

stefanv replied:
Yes! Usually, I do too :)

What I'd love is a linter that is able to break on newlines, but unwrap each line. I've been searching for a long time but without success.

stefanv replied:
Hey, presto :magic_wand:

nabobalis replied:
Actually if someone does know of a linter or program that does that, I would love to be told.

Thanks @stefanv!

jarrodmillman replied:
Feel free to continue the discussion here: https://github.com/scientific-python/specs/issues/54

bsipocz reviewed (approved):
I love this post as is 💖 

Maybe also add the other group picture we took by the late, at least the previous summit post had multiple photos.

The major work thing that many of us were side tracking during the summit was submitting the NASA proposals, but I don't think that needs to be mentioned.

stefanv commented:
> Maybe also add the other group picture we took by the late, at least the previous summit post had multiple photos.

Sure! Do you have a copy? @juanis2112?

QuLogic commented on content/posts/scientific-python/dev-summit-2/index.md:
```diff
@@ -0,0 +1,97 @@
+### SciPy
+
+Several of the SciPy developers were present, and we used the opportunity to celebrate Dan Schult joining as a core developer.
+Matt and Pamphile did some work on the new distribution infrastructure, Dan worked on sparse (remotely with CJ), and newly-supported [`const` statements were added to Cython code](https://github.com/scipy/scipy/pull/20891) got reviewed and merged.
```
This last entry (and new-supported `const` statements ...) has two verbs ('were added' and 'got reviewed and merged').

stefanv replied:
Thanks, fixed!

QuLogic commented on content/posts/scientific-python/dev-summit-2/index.md:
```diff
@@ -0,0 +1,97 @@
+
+### Unplanned collaborations
+
+As is the nature of these events, some collaborations arise spontaneously.
+E.g., Nick and Ariel explored using Awkward Array for neuro-tractography, while Nick and Mridul explored using [scipp](https://scipp.github.io/index.html) for high-energy physics data.
+Guen worked on telemetry, while Inessa and Sanket discussed community surveys and governance.
+Sebastian and Thomas [discussed parallelization APIs](https://hackmd.io/84thx0ucQ2ab17ZYrBhWRw).
```
`e.g.` is odd at the beginning of a sentence to me. Are all of these lines examples of the collaborations? In which case, should it be a bullet list?
```suggestion
As is the nature of these events, some collaborations arise spontaneously. For example,
- Nick and Ariel explored using Awkward Array for neuro-tractography, while Nick and Mridul explored using [scipp](https://scipp.github.io/index.html) for high-energy physics data.
- Guen worked on telemetry, while Inessa and Sanket discussed community surveys and governance.
- Sebastian and Thomas [discussed parallelization APIs](https://hackmd.io/84thx0ucQ2ab17ZYrBhWRw).
```

stefanv replied:
I think this is OK, but I'll let others weigh in; let's see.

dschult replied:
To me, it seems strange to have bullet lists with more than one item in each bullet.  And splitting the sentences further loses the conversational feel of the blog. I suppose we could spell out whatever “e.g.” stands for, but “that’s Greek to me” (or maybe Latin?).  :)  While “e.g” and “for example” are literally translations of each other, the abbreviation feature means that “e.g.”  seems to be used in the middle of sentences rather than the beginning. (We’re not supposed to start a sentence with an abbreviation.)

How about starting with “For example,” instead of “E.g.,”

bsipocz replied:
Controversial suggestion, I know, but I feel it so often in many repos that all of them should use the no-bad-latin CI job we have over at the turning way ;) https://github.com/numpy/numpy-tutorials/blob/main/content/mooreslaw-tutorial.md?plain=1#L480-L496

stefanv replied:
Expanded.

stefanv replied:
Because the list now got long, the listification happened anyway.

QuLogic commented on content/posts/scientific-python/dev-summit-2/index.md:
```diff
@@ -0,0 +1,97 @@
+But, besides the inherent satisfaction of working together with this great group, the best feature of the summit was that we were able to hang out, bonding over our communal joys and struggles—both technical and personal.
+
+We are grateful to the ecosystem developers who gave up their time to attend the summit (many had to put in leave _just to do more work_!).
+The summits are valuable, and translate to a lot of work work getting done and decisions being made.
```
Unless this was here for emphasis:
```suggestion
The summits are valuable, and translate to a lot of work getting done and decisions being made.
```

stefanv replied:
Added in latest commit, and included you as author.

nabobalis reviewed (approved):
Thanks for putting this together @stefanv

dschult reviewed (approved):
This is really well done Stefan!  Your love for the group and the event and the individuals shows itself despite being implicit.  Huzzah!

jarrodmillman reviewed (approved):
This looks great!

rossbar reviewed (approved):
Indeed, thanks @stefanv - this is a great summary. These summits are always fantastic - can't wait for the next one!

melissawm reviewed (commented):
This looks great, thanks for the writeup @stefanv !

drammock commented on content/posts/scientific-python/dev-summit-2/index.md:
```diff
@@ -0,0 +1,98 @@
+
+In June, several of us got together for the "annual" (well, we've had it twice now) Scientific Python developer summit in Seattle.
+Our friends at the eScience Institute were again kind enough to host us.
+This time around, we made the event a bit shorter: a week is a long time to take off, and it also gets a bit overstimulating/exhausting, given the amount of conversation and work that takes place.
```
This sentence makes me hesitate a bit; I *liked* the full week format of the first one, and found 3 days felt a bit short. I worry that saying "5 days was exhausting" might make it harder to get funding for a full week if we decided to go back to that format. 

(Also I thought part of the complication was overlapping with commencement --- not that I think we should mention that, but it feels a bit disingenuous to only say "we shortened because 5 days was too long"). Would it be OK to just say "this time was shorter" without explanation, or with a vague explanation like "for logistical reasons"?

```suggestion
This time around, we made the event a bit shorter (for logistical reasons).
```

stefanv replied:
I noted that we shortened it to avoid clashing with graduation.

drammock commented on content/posts/scientific-python/dev-summit-2/index.md:
```diff
@@ -0,0 +1,98 @@
+Documentation was a much more popular topic than anticipated!
+
+- The new [mystmd](https://mystmd.org/guide) tooling generated some excitement, and an [experimental port of the NumPy tutorials](https://github.com/numpy/numpy-tutorials/tree/mystjs) was made.
+- Recommendations on consistent use of [backticks](https://github.com/numpy/numpydoc/pull/525) and [monospaced font](https://github.com/pydata/pydata-sphinx-theme/issues/1852) were submitted to numpydoc and pydata-sphinx-theme, respectively.
```
This may be too much detail / too in-the-weeds, feel free to trim down:

```suggestion
- Recommendations on consistent use of [backticks](https://github.com/numpy/numpydoc/pull/525) and [monospaced font](https://github.com/pydata/pydata-sphinx-theme/issues/1852) were submitted to numpydoc and pydata-sphinx-theme, respectively.
- Eric isolated [a non-deterministic bug in Sphinx](https://github.com/sphinx-doc/sphinx/issues/12409) that was affecting SciPy's documentation builds, and found a work-around for SciPy that had been eluding several of us for months!
- Madicken, Paul, and Dan [worked together](https://github.com/pydata/pydata-sphinx-theme/pull/1861) to extend PyData Sphinx Theme's testing infrastructure, by combining Sphinx Build Factory (to generate small test sites) with Playwright (to mock interactions with those sites during testing).
```

stefanv replied:
I split these items: sent Eric's item to SciPy, and condensed the Sphinx testing item slightly. Please take a look!

drammock replied:
> I split these items: sent Eric's item to SciPy, and condensed the Sphinx testing item slightly. Please take a look!

LGTM!

drammock reviewed (approved):
Great write-up. 
Notably absent is any mention of matplotlib, @ksunden and @QuLogic any highlights you want to add?

lagru reviewed (approved):
Thanks @stefanv. Reads nicely!

stefanv commented:
Does anyone have a photo of Sophia trashing us in chess? Perhaps @ivanov?

drammock commented:
> Does anyone have a photo of Sophia trashing us in chess? Perhaps @ivanov?

There's one photo here: https://discord.com/channels/786703927705862175/1235328192294817912/1247747552720322600

there's a google drive folder linked from https://discord.com/channels/786703927705862175/1235328192294817912/1247750578898210819 (I never requested access, as I never took any pictures) but maybe more in there?

stefanv commented:
Thanks, I have those photos; I was hoping to get a nice action shot, let's see what crops up.

stefanv commented:
@larsoner @seberg @MridulS You approved, but can I also add you as authors?

larsoner commented:
Yes feel free!

MSanKeys963 commented on content/posts/scientific-python/dev-summit-2/index.md:
```diff
@@ -0,0 +1,109 @@
+
+- [SPEC-?: Dispatching (`spatch`)](https://hackmd.io/yI1iAqekQIq0a4jLS9WPyw)
+- [SPEC-8: Securing The Release Process](https://scientific-python.org/specs/spec-0008/)
+- [SPEC-9: Governance](https://github.com/scientific-python/specs/pull/323)
```
```suggestion
- [SPEC-9: Governance](https://scientific-python.org/specs/spec-0009/)
```
Minor comment: The governance PR was recently merged, so maybe we should link the SPEC document on the website instead of PR.

MSanKeys963 commented:
Thanks for working on this, @stefanv.

Checks the 3C list: Clear, concise and crisp. 😄

stefanv commented on content/posts/scientific-python/dev-summit-2/index.md:
```diff
@@ -0,0 +1,106 @@
+
+As is the nature of these events, some collaborations arise spontaneously.
+For example, Nick and Ariel explored using Awkward Array for neuro-tractography, while Nick and Mridul explored using [scipp](https://scipp.github.io/index.html) for high-energy physics data.
+Guen worked on telemetry, while Inessa and Sanket discussed community surveys and governance.
```
could I just make it "best practices for community surveys and project governance"?

InessaPawson commented on content/posts/scientific-python/dev-summit-2/index.md:
```diff
@@ -0,0 +1,109 @@
+As is the nature of these events, some collaborations arise spontaneously.
+For example, Nick and Ariel explored using Awkward Array for neuro-tractography, while Nick and Mridul explored using [scipp](https://scipp.github.io/index.html) for high-energy physics data.
+Guen worked on telemetry, while Inessa and Sanket discussed community surveys and governance.
+Sebastian and Thomas [discussed parallelization APIs](https://hackmd.io/84thx0ucQ2ab17ZYrBhWRw).
```
```suggestion
And with input from Tim and Thomas, Inessa was able to finalize the design of the 2024 scikit-learn user survey questionnaire. 
Sebastian and Thomas [discussed parallelization APIs](https://hackmd.io/84thx0ucQ2ab17ZYrBhWRw).
```

stefanv replied:
Added, after moving items to list.

dschult commented on content/posts/scientific-python/dev-summit-2/index.md:
```diff
@@ -0,0 +1,116 @@
+- Inessa and Sanket discussed best practices for community surveys and project governance.
+- Sebastian and Thomas [discussed parallelization APIs](https://hackmd.io/84thx0ucQ2ab17ZYrBhWRw).
+- Inessa, with input from Tim and Thomas, finalized the design of the 2024 scikit-learn user survey.
+
```
```suggestion
- Erik and Dan discussed index compression options for CSR-like nD sparse arrays.

```

QuLogic commented:
> Notably absent is any mention of matplotlib, @ksunden and @QuLogic any highlights you want to add?

Hmm, definitely documentation-related items: @larsoner and I fixed some intersphinx issues in sphinx-gallery, @steppi and I discussed translations for the Matplotlib website, and I discussed `sphinx-tags` with @melissawm.

We also endorsed some SPECs on behalf of the Matplotlib project.

stefanv commented:
@QuLogic Do you want to propose a sentence or two to add?

stefanv commented:
I've added the matplotlib updates, and I've updated the post date to tomorrow.

If someone can take a last look and merge tomorrow (30 September), I'd appreciate it.

nabobalis commented:
Thank you @stefanv and all the reviewers!

stefanv commented:
If anyone feels strongly about the chess photo, let me know and I'll add it as an update :)

bsipocz commented:
FWIW, I would love to see a chess photo, or chess being mentioned :)

stefanv commented:
> FWIW, I would love to see a chess photo, or chess being mentioned :)

Photo added in https://github.com/scientific-python/blog.scientific-python.org/pull/239
